hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/s3guard.md

```diff
@@ -971,6 +971,111 @@ from previous days, and and choosing a combination of
 retry counts and an interval which allow for the clients to cope with
 some throttling, but not to time out other applications.
 
+## Read-After-Overwrite Consistency
+
+S3Guard provides read-after-overwrite consistency through ETags (default) or
+object versioning checked either on the server (default) or client. This works
+such that a reader reading a file after an overwrite either sees the new version
+of the file or an error. Without S3Guard, new readers may see the original
+version. Once S3 reaches eventual consistency, new readers will see the new
+version.
+
+Readers using S3Guard will usually see the new file version, but may
+in rare cases see `RemoteFileChangedException` instead. This would occur if
+an S3 object read cannot provide the version tracked in S3Guard metadata.
+
+S3Guard achieves this behavior by storing ETags and object version IDs in the
+S3Guard metadata store (e.g. DynamoDB). On opening a file, S3AFileSystem
+will look in S3 for the version of the file indicated by the ETag or object
+version ID stored in the metadata store. If that version is unavailable,
+`RemoteFileChangedException` is thrown. Whether ETag or version ID and
+server or client mode is used is determed by the
+[fs.s3a.change.detection configuration options](./index.html#Handling_Read-During-Overwrite).
```

Comment:
If we modify or add new behavior of this variable, then we should update the description in core-default.xml.
This will patch will change the behavior because after this s3guard will throw an exception based on how this property is set.

Reply:
Good call.  I'll take care of this soon.

Reply:
Updated in https://github.com/apache/hadoop/pull/675/commits/5f0532b5666cd5b855e219d4ba02cbbfe19c33c2

Let me know if you think that change is enough.  I considered somehow referencing documentation in index.md and/or s3guard.md but I wasn't sure of the best way to call that out.  A hyperlink doesn't seem feasible so I'm not sure what to call that doc in a reference.  I assume readers will probably just search for the config keys by their names and come across that other doc that way.